openid-sharedsignals-framework-1_0.md

```diff
@@ -397,14 +397,107 @@ below MAY define certain members within a Complex Subject to be Critical. A SSF
+The "exp" claim MUST NOT be used in SSF SETs.
+
+The purpose is defense in depth against confusion with other JWTs, as described
+in Sections 4.5 and 4.6 of {{RFC8417}}.
+
+### The "aud" Claim {#aud-claim}
+The "aud" claim can be a single string or an array of strings. Values that
+uniquely identify the Receiver to the Transmitter MAY be used, if the two parties
+have agreement on the format.
+
+More than one value can be present if the corresponding Receivers are known to
+the Transmitter to be the same entity, for example a web client and a mobile
+client of the same application. All the Receivers in this case MUST use the
+exact same delivery method.
+
+If multiple Receivers have the exact same delivery configuration but the
+Transmitter does not know if they belong to the same entity then the Transmitter
+SHOULD issue distinct SETs for each Receiver and deliver them separately. In
+this case the multiple Receivers might use the same service to process SETs, and
+this service might reroute SETs to respective Receivers, an "aud" claim with
+multiple Receivers would lead to unintended data disclosure.
+
+~~~ json
+{
+  "jti": "123456",
+  "iss": "https://transmitter.example.com",
+  "aud": ["receiver.example.com/web", "receiver.example.com/mobile"],
+  "iat": 1493856000,
+  "txn": 8675309,
+  "sub_id": {
+    "format": "opaque",
+    "id": "72e6991badb44e08a69672960053b342"
+  },
+  "events": {
+    "https://schemas.openid.net/secevent/ssf/event-type/verification": {
+      "state": "VGhpcyBpcyBhbiBleGFtcGxlIHN0YXRlIHZhbHVlLgo="
+    }
+  }
+}
+~~~
+{: title="Example: SET with array 'aud' claim" #figarrayaud}
+
+### The "txn" claim {#txn-claim}
+Transmitters SHOULD set the "txn" claim value in Security Event Tokens (SETs). If the value is present, it MUST be unique to the underlying event that caused the Transmitter to generate the Security Event Token (SET). The Transmitter, however, may use the same value in the "txn" claim across different Security Events Tokens (SETs), such as session revoked and credential change, to indicate that the SETs originated from the same underlying cause or reason.
+
+## Event Properties {#event-properties}
+
+### The "events" claim {#events-claim}
+The "events" claim SHOULD contain only one event. Multiple event type URIs are
+permitted only if they are alternative URIs defining the exact same event type. The type of the event is specified by the key in the value of the `events` claim. The value of this field is the event object.
+
+### Event type specific fields
+The event object inside the `events` claim MAY have one or more fields that are uniquely determined by the type of the event.
+
+### The "event_data" field {#event-data-claim}
+Any fields that the Transmitter wishes to communicate to the receiver, but which are not specified by the specific event type, SHOULD be placed in a `event_data` field inside the event object. If this field is present, then its value MUST be a JSON object.
```

Comment:
- "event" part of "event_data" redundant, maybe metadata?
- How do we prevent bloating the SETs?

Reply:
Please comment on issue #221 for questions about the idea itself. I'll remove the event_data related text from this PR and keep it only about refactoring.

Reply:
Removed text about event_data from this PR